wallet/src/de/schildbach/wallet/ui/dashpay/NotificationsAdapter.kt

```diff
@@ -268,4 +307,135 @@ class NotificationsAdapter(val onContactRequestButtonClickListener: OnContactReq
         fun onAcceptRequest(usernameSearchResult: UsernameSearchResult, position: Int)
         fun onIgnoreRequest(usernameSearchResult: UsernameSearchResult, position: Int)
     }
+
+    private val transactionCache: HashMap<Sha256Hash, TransactionCacheEntry> = HashMap()
+
+    private class TransactionCacheEntry constructor(val value: Coin, val sent: Boolean, val self: Boolean, val showFee: Boolean, val address: Address?,
+                                                            val addressLabel: String?, val type: Transaction.Type)
+
+    inner class TransactionViewHolder (inflater: LayoutInflater, parent: ViewGroup) : ViewHolder(R.layout.notification_transaction_row, inflater, parent) {
```

Comment:
This was copy/pasted from the TransactionAdapter.java class.  Ideally this should be refactored into a separate file that can be used on the Home Screen and any Notifications screen.